### DISCUSSION TITLE
Sort open buffers?

### DISCUSSION THREAD

**Discussion by yanniskatsaros** [helix-editor/helix#11123] [Q&A]
## Is there a way to sort open buffers in the editor (by name, file extension, etc.)?

For example, if I had the following buffers open:

| c-example.py | b-example.toml | a-example.py | d-example.txt | e-example.py |
|:------------:|:--------------:|:------------:|:-------------:|:------------:|

I was hoping there would be something akin to doing:

```
:buffer-sort name
```

which would sort the buffers to:

| a-example.py | b-example.toml | c-example.py | d-example.txt | e-example.py |
|:------------:|:--------------:|:------------:|:-------------:|:------------:|

or instead by file extension:

```
:buffer-sort extension
```

| a-example.py | c-example.py | e-example.py | b-example.toml | d-example.txt |
|:------------:|:--------------:|:------------:|:-------------:|:------------:|

I tried exploring suggestions from the command palette using `<space>?` but didn't see anything relevant.

**Comment by grv07** [comment]
What is your use case?
Generally  `Space+b`  resolve most of mines.

**Reply by yanniskatsaros** [reply]
@grv07 my use case is that I usually have many buffers open at once and I have keymaps setup (`Shift+h` to move to the left buffer and `Shift+l` to move to the right buffer) to switch between them quickly. There are situations when I'd like to logically group buffers together to make it easier when I'm switching back and forth between them.

Not a huge deal if this functionality isn't available, I just wanted to ask in case I was missing something. `Space+b` is definitely useful and I do use it, this is just a slightly different use case.

**Comment by giuliano-macedo** [comment]
Relates to https://github.com/helix-editor/helix/discussions/10278
